kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/SubscriptionServiceImpl.java

```diff
@@ -76,17 +78,17 @@ private void createSubscriptionPack(String name, SubscriptionPackType type, int
             }
         }
 
-        subscriptionPackDataService.create(new SubscriptionPack(name, type, messagesPerWeek, messages));
+        subscriptionPackDataService.create(new SubscriptionPack(name, type, weeks, messagesPerWeek, messages));
     }
 
 
     @Override
     public void createSubscription(long callingNumber, Language language, SubscriptionPack subscriptionPack,
                                    SubscriptionMode mode) {
-        Subscriber subscriber = subscriberService.getSubscriber(callingNumber);
+        Subscriber subscriber = subscriberDataService.findByCallingNumber(callingNumber);
         if (subscriber == null) {
             subscriber = new Subscriber(callingNumber, language);
-            subscriberService.add(subscriber);
+            subscriberDataService.create(subscriber);
         }
 
         if (mode == SubscriptionMode.IVR) {
```

Comment:
I think you have to merge master since @frankhuster changed SubscriptionMode to SubscriptionOrigin